File: apps/android/feature/canvas/domain/src/main/kotlin/com/scribblefit/feature/canvas/domain/GetScribblesForDateUseCase.kt
package com.scribblefit.feature.canvas.domain

import com.scribblefit.core.model.Scribble
import com.scribblefit.feature.scribble.domain.ScribbleRepository
import kotlinx.coroutines.flow.Flow
import java.time.LocalDate
import java.time.ZoneId

class GetScribblesForDateUseCase(
    private val repository: ScribbleRepository
) {
    operator fun invoke(date: LocalDate): Flow<List<Scribble>> {
        val timestamp = date.atStartOfDay(ZoneId.systemDefault()).toInstant().toEpochMilli()
        return repository.getScribblesByDate(timestamp)
    }
}

File: apps/ios/ScribbleFit/LocalPackages/Features/FeatureScribble/Sources/FeatureScribble/Domain/UseCases/GetScribblesForDateUseCase.swift
import Foundation
import CoreModel

@MainActor
public struct GetScribblesForDateUseCase {
    private let repository: ScribbleRepository

    public init(repository: ScribbleRepository) {
        self.repository = repository
    }

    public func execute(date: Date) -> AsyncStream<[Scribble]> {
        repository.observeScribbles(for: date)
    }
}
